bot/seasons/christmas/adventofcode.py

```diff
@@ -170,10 +170,21 @@ def __init__(self, bot: commands.Bot):
         """Return time left until next day."""
         if not is_in_advent():
             datetime_now = datetime.now(EST)
-            december_first = datetime(datetime_now.year + 1, 12, 1, tzinfo=EST)
-            delta = december_first - datetime_now
+
+            # Calculate the delta to this & next year's December 1st to see which one is closest and not in the past
+            this_year = datetime(datetime_now.year, 12, 1, tzinfo=EST)
+            next_year = datetime(datetime_now.year + 1, 12, 1, tzinfo=EST)
+            deltas = (dec_first - datetime_now for dec_first in (this_year, next_year))
+            delta = min(delta for delta in deltas if delta >= timedelta())  # timedelta() gives 0 duration delta
```

Comment:
Why is `if delta >= timedelta()` needed? Is this for the edge case that the delta is calculated on the exact moment the event starts?

Reply:
To filter negative timedeltas

Reply:
Never mind. I thought the event lasted all of December so negatives would not be possible in practice. But it only goes until the 25th.